### DISCUSSION TITLE
Google-style additional login challenges for suspicious logins

### DISCUSSION THREAD

**Discussion by tyteen4a03** [ory/kratos#2811] [Q&A]
Does Ory Kratos have built-in support for suspicious login blocking (e.g. location-based checks ala Google) or are there any plugins that can do the same thing?

**Comment by vinckr** [comment]
Hello @tyteen4a03,
We recently implemented a feature that logs IP Address, UA & Location (if available): https://github.com/ory/kratos/pull/2715
@kelkarajay
